### PR TITLE
Add visit method in transitivevisitor for InterfaceDeclaration

### PR DESCRIPTION
**Pull request by lucica28**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Review by RazvanN7 — APPROVED** [review]

File: compiler/src/dmd/transitivevisitor.d [hunk 1]
```diff
@@ -799,6 +799,15 @@ package mixin template ParseVisitMethods(AST)
                 s.accept(this);
     }
 
+     override void visit(AST.InterfaceDeclaration d)
```
Comment (RazvanN7) [inline compiler/src/dmd/transitivevisitor.d:802]:
```suggestion
    override void visit(AST.InterfaceDeclaration d)
```

**Review by RazvanN7 — COMMENTED** [review]
Why is this needed? InterfaceDeclaration inherits ClassDeclaration which is treated above. What's the point in copy pasting this code?

**Comment by lucica28** [comment]
> Why is this needed? InterfaceDeclaration inherits ClassDeclaration which is treated above. What's the point in copy pasting this code?

This is needed when you want to extend this class and you want to override the visit method only for class declaration. With the current implementation if you override the visit method of ClassDeclaration you automatically override the visit logic for InterfaceDeclaration. Now for sure in the class that extends the transitive visitor you could override both ClassDeclaration and InterfaceDeclaration but I thought it would be cleaner this way.(with a visit method also for InterfaceDeclaration in transitivevisitor.d)

**Comment by RazvanN7** [comment]
@lucica28 Ok. Just mind the comment.

**Comment by maxhaton** [comment]
@lucica28 ready?

**Comment by lucica28** [comment]
> @lucica28 ready?

Yea, I would say this can be merged

**Pull request merged**